src/middlewared/middlewared/plugins/jail.py

```diff
@@ -631,6 +664,29 @@ def list_resource(self, resource, remote, want_cache, branch):
 
         return resource_list
 
+    @periodic(interval=86400)
+    @private
+    @accepts(Str('branch', null=True, default=None))
+    @job(lock='retrieve_plugin_versions')
+    def retrieve_plugin_versions(self, job, branch=None):
+        branch = branch or self.get_version()
+        ioc_plugin = IOCPlugin(branch=branch)
+        if self.get_activated_pool():
+            plugins = ioc_plugin.fetch_plugin_versions()
+        else:
+            with tempfile.TemporaryDirectory() as td:
+                github_repo = 'https://github.com/freenas/iocage-ix-plugins.git'
+                try:
+                    IOCPlugin._clone_repo(branch, github_repo, td, depth=1)
+                except Exception:
+                    return {}
```

Comment:
Shouldn't we raise something here instead? @william-gr?

Reply:
I am on the fence between logging and raise

Reply:
@skarekrow logging seems better in my opinion.

Reply:
Agreed as well

Reply:
Perfect